src/EditorFeatures/CSharpTest2/Recommendations/VoidKeywordRecommenderTests.cs

```diff
@@ -727,5 +727,60 @@ void M(Action<int> a)
     }
 }");
         }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInFunctionPointer01()
+        {
+            await VerifyKeywordAsync(@"
+class C
+{
+    delegate*<$$");
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInFunctionPointer02()
+        {
+            await VerifyKeywordAsync(@"
+class C<T>
+{
+    C<delegate*<$$");
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInFunctionPointer03()
+        {
+            await VerifyKeywordAsync(@"
+class C
+{
+    delegate*<int, $$");
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInFunctionPointer04()
+        {
+            await VerifyKeywordAsync(@"
+class C
+{
+    delegate*<int, v$$");
+        }
+
+        [Theory]
+        [InlineData("ref")]
+        [InlineData("ref readonly")]
+        [InlineData("readonly")]
+        [InlineData("out")]
+        [InlineData("in")]
+        [Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInFunctionPointerAfterRefKind(string refKind)
+        {
+            await VerifyAbsenceAsync($@"
+class C
+{{
+    delegate*<int, {refKind} $$");
```

Comment:
can't have `ref void**`?

Reply:
Thats... a good point. Will certainly make the code simpler.